php/Connector.php

```diff
@@ -7,14 +7,25 @@ class Connector
     private $sock;
     private $jdbc_enc;
     private $app_enc;
+    private $host;
+    private $port;
+    private $socketHandler;
 
     public $last_search_length = 0;
 
-    function __construct($host = "localhost", $port = "4444", $jdbc_enc = "ascii", $app_enc = "ascii")
+    function __construct($host = "localhost", $port = "4444", $jdbc_enc = "ascii", $app_enc = "ascii", $socketHandler = null)
     {
-        $this->sock = fsockopen($host, $port);
+        $this->socketHandler = $socketHandler ?? 'fsockopen';
```

Comment:
Just set the default to fsockopen: `function __construct($host = "localhost", $port = "4444", $jdbc_enc = "ascii", $app_enc = "ascii", $socketHandler = 'fsockopen')`

No need to do this null check then.